core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java

```diff
@@ -466,13 +471,24 @@ private void connectBlock(final Block block, StoredBlock storedPrev, boolean exp
             }
             if (expensiveChecks && block.getTimeSeconds() <= getMedianTimestampOfRecentBlocks(head, blockStore))
                 throw new VerificationException("Block's timestamp is too early");
-            
+
+            // BIP 66: Enforce block version 3 once it's a supermajority of blocks
+            if (block.getVersion() == Block.BLOCK_VERSION_BIP34) {
+                final Integer count = versionTally.getCount(Block.BLOCK_VERSION_BIP66);
+                if (count != null
```

Comment:
I like my return values to throw exceptions if you forget to check them. Want me to change it to an int?